docs/usage/00-query-processing.md

```diff
@@ -0,0 +1,495 @@
+    },
+    {
+      "op": "in",
+      "content": {
+        "fieldName": "data.stage",
+        "value": ["Stage IV"]
+      }
+    },
+    {
+      "op": "in",
+      "content": {
+        "fieldName": "data.vital_status",
+        "value": ["Deceased"]
+      }
+    }
+  ]
+}
+```
+
+This SQON filters for records where:
+- Primary site is "Brain" **AND**
+- Stage is "Stage IV" **AND**
+- Vital status is "Deceased"
+
+</details>
+
+The client sends the SQON to Arranger as part of a GraphQL request:
+
+<details>
+<summary><b>B) View the GraphQL query with embedded SQON</b></summary>
+
+```json showLineNumbers
+{
+  "query": "query tableData($sqon: JSON, ...) { file { hits(filters: $sqon, ...) { ... } } }",
+  "variables": {
+    "sqon": {
+      "op": "and",
+      "content": [
+        {
+          "op": "in",
+          "content": {
+            "fieldName": "data.primary_site",
+            "value": ["Brain"]
+          }
+        },
+        {
+          "op": "in",
+          "content": {
+            "fieldName": "data.stage",
+            "value": ["Stage IV"]
+          }
+        },
+        {
+          "op": "in",
+          "content": {
+            "fieldName": "data.vital_status",
+            "value": ["Deceased"]
+          }
+        }
+      ]
+    },
+    "first": 20,
+    "offset": 0,
+    "sort": []
+  }
+}
+```
+
+**Key Components:**
+
+- **SQON Filter** (lines 4-29): The filter logic passed as a query variable
+- **`filters: $sqon`** (line 2): The GraphQL query parameter that receives the filter
+- **Pagination** (lines 24-25): `first` and `offset` control how many records to fetch and where to start
+- **Sorting** (line 26): Order of results (empty in this example)
+
+</details>
+
+Arranger receives the GraphQL request and translates the entire query, including the embedded SQON filter, into a native Elasticsearch query:
+
+<details>
+<summary><b>C) View the resulting Elasticsearch query</b></summary>
+
+```json showLineNumbers
+{
+  "query": {
+    "bool": {
+      "must": [
+        {
+          "terms": {
+            "data.primary_site": ["Brain"]
+          }
+        },
+        {
+          "terms": {
+            "data.stage": ["Stage IV"]
+          }
+        },
+        {
+          "terms": {
+            "data.vital_status": ["Deceased"]
+          }
+        }
+      ]
+    }
+  },
+  "from": 0,
+  "size": 20,
+  "sort": []
+}
+```
+
+**Translation Breakdown:**
+
+| SQON Component | Elasticsearch Equivalent | Notes |
+|----------------|-------------------------|-------|
+| `"op": "and"` | `"bool": { "must": [...] }` | AND logic requires all conditions to match |
+| `"op": "in"` | `"terms": { "field": [...] }` | IN operation uses Elasticsearch's `terms` query |
+| `"fieldName": "data.primary_site"` | `"data.primary_site": [...]` | Field name maps directly to the index mapping |
+| `"first": 20` | `"size": 20` | Pagination size parameter |
+| `"offset": 0` | `"from": 0` | Pagination offset parameter |
+
```

Comment:
prettier generally does a good job at inserting spaces in md to make tables readable even without a md render. wonder if the settings in this repo need adjustment to make sure this tool is applied here